metrics/web/test/project_groups/presentation/widgets/project_checkbox_list_test.dart

```diff
@@ -0,0 +1,135 @@
+import 'package:metrics/project_groups/presentation/state/project_groups_notifier.dart';
+import 'package:metrics/project_groups/presentation/view_models/project_checkbox_view_model.dart';
+import 'package:metrics/project_groups/presentation/widgets/project_checkbox_list.dart';
+import 'package:metrics/project_groups/presentation/widgets/project_checkbox_list_tile.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../../test_utils/project_groups_notifier_mock.dart';
+import '../../../test_utils/test_injection_container.dart';
+
+void main() {
+  group("ProjectCheckboxList", () {
+    testWidgets(
+      "displays all given project checkbox view models",
+      (WidgetTester tester) async {
+        final projectGroupsNotifier = ProjectGroupsNotifierMock();
+
+        const projectCheckboxViewModels = [
+          ProjectCheckboxViewModel(isChecked: false, id: 'id', name: 'name'),
+          ProjectCheckboxViewModel(isChecked: false, id: 'id', name: 'name'),
+          ProjectCheckboxViewModel(isChecked: false, id: 'id', name: 'name'),
+        ];
+        when(projectGroupsNotifier.projectCheckboxViewModels)
+            .thenReturn(projectCheckboxViewModels);
+
+        await tester.pumpWidget(_ProjectCheckboxListTestbed(
+          projectGroupsNotifier: projectGroupsNotifier,
+        ));
+
+        final projectCheckboxWidgets = tester.widgetList(
+          find.byType(ProjectCheckboxListTile),
+        );
+
+        expect(
+          projectCheckboxWidgets.length,
+          equals(projectCheckboxViewModels.length),
+        );
+      },
+    );
+
+    testWidgets(
+      "displays the text placeholder with an error message if there is a projects error",
+      (WidgetTester tester) async {
+        final projectGroupsNotifier = ProjectGroupsNotifierMock();
+        const error = 'error';
+
+        when(projectGroupsNotifier.projectsErrorMessage).thenReturn(error);
+
+        await tester.pumpWidget(
+          _ProjectCheckboxListTestbed(
+            projectGroupsNotifier: projectGroupsNotifier,
+          ),
+        );
+
+        final textPlaceholderFinder = find.widgetWithText(
+          TextPlaceholder,
+          error,
+        );
+
+        expect(textPlaceholderFinder, findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "displays the loading placeholder if a list of project checkbox view models is null",
+      (WidgetTester tester) async {
+        final projectGroupsNotifier = ProjectGroupsNotifierMock();
+        when(projectGroupsNotifier.projectCheckboxViewModels).thenReturn(null);
+
+        await tester.pumpWidget(
+          _ProjectCheckboxListTestbed(
+            projectGroupsNotifier: projectGroupsNotifier,
+          ),
+        );
+
+        expect(find.byType(LoadingPlaceholder), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      "displays the no configured projects placeholder if project checkbox view models are empty",
```

Comment:
How do we distinguish between 'no projects configured' and 'no search results' if we use a filter?

Reply:
Oh, it seems like we've missed this. I've created #360 and #361 issues to add this handling. Thanks!